Marlin/src/inc/SanityCheck.h

```diff
@@ -1965,6 +1965,8 @@ static_assert(hbm[Z_AXIS] >= 0, "HOMING_BUMP_MM.Z must be greater than or equal
 // Z homing direction and plug usage flags
 #if Z_HOME_DIR < 0 && NONE(USE_ZMIN_PLUG, HOMING_Z_WITH_PROBE)
   #error "Enable USE_ZMIN_PLUG when homing Z to MIN."
+#elif BOTH(USE_ZMAX_PLUG, HOMING_Z_WITH_PROBE)
```

Comment:
I am not entirely sure whether there is a use-case where this combination is valid. I don't completely understand how these USE_*_PLUG options are used when using multiple endstops, for example.

Reply:
The "`USE_blah_PLUG`" options only exist to tell `pins.h` to leave the defines for those endstops intact rather than killing them with `#undef`. In the past this was needed because there were conditionals taking the existence of some pins as signaling that they should be initialized during startup and …maybe… that they should be monitored as endstops.

It is quite possible now that we have enough conditionals and enough confidence to be able to get rid of these defines and to use other methods to figure out when we need the pins to be treated as endstops, probes, etc.

Reply:
OTOH, although the default `USE_blah_PLUG` defines _could_ be automated based on enabled endstop- and probe-related options, the `USE_blah_PLUG` defines _might_ also be used by someone who wants to be able to refer to "the `Z_MIN_PIN` of the active board" by name even though they are not using the pin for that purpose.

Anyway, if we were interested in making the endstop pin define _always_ refer to the physical endstop pins on all the boards, and then adding an intermediate `*_ENDSTOP_PIN` for each axis that wants one, that would add some flexibility in pin choice. It would be tedious to build this extra layer, but it would also make it more clear in the code when a pin is really being used as an endstop.

Reply:
I guess since you decided to close this I will need to try to fix the endstop code that behaves improperly, instead of just avoiding the combination.